server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -52,6 +52,12 @@
      * The generation value indicating the repository has no index generational files.
      */
     public static final long EMPTY_REPO_GEN = -1L;
+
+    /**
+     * The generation value indicating that the repository generation is unknown.
+     */
+    public static final long UNKNOWN_REPO_GEN = -2L;
```

Comment:
I needed something to signifiy the "unknown" state. Previously this was done by just `null` in the separately kept map for the repo states. But since I didn't want to have a `Nullable` `LONG` field I went with this.